app/browse/data-table-preview.tsx

```diff
@@ -111,20 +120,19 @@ export const DataTablePreview = ({
           {sortedObservations.map((obs, i) => {
             return (
               <TableRow key={i}>
-                {sortedComponents.map((component) => {
-                  const format = formatters[component.id];
+                {sortedComponents.map((c) => {
+                  const numerical = isNumericalMeasure(c);
+                  const format = formatters[c.id];
+                  const cu = conversionUnitsByComponentId?.[c.id];
+                  const v = obs[c.id];
 
                   return (
                     <TableCell
-                      key={component.id}
+                      key={c.id}
                       component="td"
-                      sx={{
-                        textAlign: isNumericalMeasure(component)
-                          ? "right"
-                          : "left",
-                      }}
+                      sx={{ textAlign: numerical ? "right" : "left" }}
                     >
-                      {format(obs[component.id])}
+                      {format(numerical && cu && v ? +v * cu.factor : v)}
```

Comment:
I wonder why the conversion is done so late, at display time. Another approach would be to change the data themselves. This way we would not have to change the formatting everywhere. It could be seen as a job for the formatter here. The formatter would be created with conversionUnits in mind and would the conversion. Otherwise we have formatting inside the formatter and then here.